pkg/dbconfig/datanode_config.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dbconfig
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/GreptimeTeam/greptimedb-operator/apis/v1alpha1"
+	"github.com/GreptimeTeam/greptimedb-operator/pkg/utils"
+)
+
+var _ Config = &DatanodeConfig{}
+
+// DatanodeConfig is the configuration for the datanode.
+type (
+	DatanodeConfig struct {
+		// Node running mode.
+		Mode string `toml:"mode,omitempty"`
```

Comment:
question: [initializer](https://github.com/GreptimeTeam/greptimedb-operator/blob/develop/cmd/initializer/types.go#L35) is define `DatanodeConfig` struct, now here define too, what kind of behavior will there be?

Reply:
The initializer will be refactored to use `dbconfig` package. The original configs will be removed. We put it in the next [PR](https://github.com/GreptimeTeam/greptimedb-operator/pull/100).